third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -1787,56 +1787,37 @@ function DOM_getAllBoundingClientRects() {
     .map((elem, i) => {
       const id = registerPlainObject(elem.raw) || i;
 
-      const { left, top, right, bottom } = shiftRect(elem.raw.getBoundingClientRect(), elem.offset);
+      // Use the bounding client rect as a fallback.
+      let { left, top, right, bottom } =
+        shiftRect(elem.raw.getBoundingClientRect(), elem.offset);
+
+      // If we can get a box-model for the element, use its border
+      // as the bounding rect.
+      try {
+        const nodeId = getBlinkNodeIdByRrpId(id);
+        const cdpModel = fromJsGetBoxModel(nodeId);
+        if (cdpModel) {
+          const top_left = cdpModel.border.slice(0, 2);
+          const top_right = cdpModel.border.slice(2, 4);
+          const bottom_right = cdpModel.border.slice(4, 6);
+          const bottom_left = cdpModel.border.slice(6, 8);
+          left = Math.min(top_left[0], bottom_left[0]);
```

Comment:
Can bottom_left[0] and top_left[0] really be different?

Reply:
I'm replacing this code using a test case, but I don't know how it will behave generically.  There are 8 points.  I reasoned that in certain transforms (e.g. rotation), the bounding box might actually not be an aligned rectangle.  I don't have a great intuition for how this works in all circumstances, so I just tried to code it defensively, including the "fallback" of the original client rect which sort of worked otherwise.

Reply:
Okay, fair enough (though if you really want to be thorough, you'll have to select min/max values from _all_ of the supplied values, since a rotation can put each x,y value anywhere in the ordering of all the points.  Can't say I'm too concerned about this, though, and it'll probably be obvious if this is wrong somehow.)